Sming/system/uart.cpp

```diff
@@ -492,16 +492,16 @@ void uart_set_debug(int uart_nr)
     switch(s_uart_debug_nr) {
     case UART0:
         system_set_os_print(1);
-        ets_install_putc1((void *) &uart0_write_char);
+        ets_install_putc1((void (*)(char)) &uart0_write_char);
```

Comment:
That should be simplified to something like:  `ets_install_putc1(uart0_write_char);`.